### PR TITLE
Improve with serialization

### LINKED ISSUE #1780: Oracle WITH Clause
I'm trying to implemente this query with querydsl-sql 4.0.8

```sql
WITH last_rfid AS (
        SELECT
            r.login,
            max(r.updated) updated
        FROM rfid r
        GROUP BY r.login)
SELECT *
FROM rfid r
    JOIN last_rfid l
        ON l.login = r.login
            AND l.updated = r.updated
```

with this code

```java
QRfid rfid = QRfid.rfid1;
QRfid lastRfid = new QRfid("last_rfid");
List<Rfid> list = queryFactory.query()
        .with(lastRfid,
                select(lastRfid.login, lastRfid.updated.max().as("updated"))
                        .from(lastRfid)
                        .groupBy(lastRfid.login))
        .select(rfid)
        .from(rfid)
        .join(lastRfid).on(rfid.login.eq(lastRfid.login), rfid.updated.eq(lastRfid.updated))
        .fetch();
```

and I'm getting

```
1	fsousa	2016-02-02 19:40:20
2	fsousa	2016-02-01 09:43:40

```
instead of
```
1	fsousa	2016-02-02 19:40:20
```

I created a small project that reproduces the problem with HSQLDB
https://github.com/natros/querydsl-with

In the output you can see that the RFID table is joining with the same table

```sql
21:26:15.289 [main] DEBUG com.querydsl.sql.AbstractSQLQuery - with last_rfid as (
select last_rfid.LOGIN, max(last_rfid.UPDATED) as updated 
from RFID last_rfid group by last_rfid.LOGIN) 
select RFID.LOGIN, RFID.RFID, RFID.UPDATED from RFID RFID 
join RFID last_rfid 
   on RFID.LOGIN = last_rfid.LOGIN 
   and RFID.UPDATED = last_rfid.UPDATED
0 | fsousa | 01-02-2016 0:00
1 | fsousa | 12-01-2016 0:00
2 | fsousa | 01-01-2016 0:00
```

### PR DESCRIPTION
**Pull request by timowest**
Fixes #1780

### CODE REVIEW

**Comment by Shredder121** [comment]
To me this looks good.

**Comment by johnktims** [comment]
Looks good to me.

File: querydsl-sql/src/main/java/com/querydsl/sql/SQLSerializer.java [hunk 1]
```diff
@@ -190,17 +192,23 @@ protected void handleJoinTarget(JoinExpression je) {
...
             final RelationalPath<?> pe = (RelationalPath<?>) je.getTarget();
             if (pe.getMetadata().getParent() == null) {
-                SchemaAndTable schemaAndTable = getSchemaAndTable(pe);
-                boolean precededByDot;
-                if (templates.isPrintSchema()) {
-                    appendSchemaName(schemaAndTable.getSchema());
-                    append(".");
-                    precededByDot = true;
+                if (withAliases.contains(pe)) {
+                    String name = pe.getMetadata().getName();
```
Comment (natros) [inline querydsl-sql/src/main/java/com/querydsl/sql/SQLSerializer.java]: it's just me or the 'name' is not being used?

Comment (Shredder121) [inline querydsl-sql/src/main/java/com/querydsl/sql/SQLSerializer.java]: Oh, I think you're right. The value however is.
But it can be cleaned up.

**Pull request merged**